appserver/web/weld-integration/src/main/java/org/glassfish/weld/services/InjectionServicesImpl.java

```diff
@@ -95,8 +97,7 @@ public InjectionServicesImpl(InjectionManager injectionMgr, BundleDescriptor con
      * @return 
      */
     private boolean isInterceptor( Class beanClass ) {
-      HashSet<String> annos = new HashSet<>();
-      annos.add( javax.interceptor.Interceptor.class.getName() );
+      final Set<String> annos = Collections.singleton(javax.interceptor.Interceptor.class.getName());
```

Comment:
This returns an _immutable_ Set. Is this intentional? If you remove the copying from the other method modifications of this set will throw an exception.

Reply:
I preffer to not modify collections passed as parameter to public API.

Reply:
Sure. Good rule.

Reply:
OK, only validScopes are not modified. There it is possible.